srtcore/core.cpp

```diff
@@ -11006,9 +11040,7 @@ bool CUDT::runAcceptHook(CUDT *acore, const sockaddr* peer, const CHandShake& hs
             }
 
             // Any other kind of message extracted. Search on.
-            length -= (next - begin);
-            begin = next;
-            if (!begin)
+            if (!NextExtensionBlock((begin), next, (length)))
```

Comment:
This function does the same thing as the replaced part, and moreover more logically correct. The check `if (!begin)` (actually `if (!next)`) should have been done first before any operation with `next` is done. Although it will at worst make a stupid value in `length` that won't be used anyway, but it's still logically wrong.